simdjson v4.6.11 for gl_test on the A53's standalone BSP (sdk/gltf), applied by make vitis
(VITIS_PATCHES, bd/gpu_test/Makefile): padded_memory_map only where there is <sys/mman.h>, which
the BSP's C library (newlib, bare metal) does not have.

diff --git a/singleheader/simdjson.h b/singleheader/simdjson.h
index 43fe09631..c4fbbda7c 100644
--- a/singleheader/simdjson.h
+++ b/singleheader/simdjson.h
@@ -4351,7 +4351,17 @@ inline std::ostream& operator<<(std::ostream& out, simdjson_result<padded_string
 #endif
 
 
+// padded_memory_map maps files with mmap: not on Windows, nor where there is no <sys/mman.h> (a C
+// library on bare metal, which has no file system)
 #ifndef _WIN32
+#ifdef __has_include
+#if __has_include(<sys/mman.h>)
+#define SIMDJSON_HAS_MMAP 1
+#endif //__has_include(<sys/mman.h>)
+#endif //__has_include
+#endif // _WIN32
+
+#if SIMDJSON_HAS_MMAP
 /**
  * A class representing a memory-mapped file with padding.
  * It is only available on non-Windows platforms, as Windows has different APIs for memory mapping.
@@ -4402,7 +4412,7 @@ private:
   const char *data{nullptr};
   size_t size{0};
 };
-#endif // _WIN32
+#endif // SIMDJSON_HAS_MMAP
 
 
 
@@ -4691,7 +4701,9 @@ inline padded_string_view pad_with_reserve(std::string& s) noexcept {
 #ifndef _WIN32
 #include <fcntl.h>
 #include <stdio.h>
+#if SIMDJSON_HAS_MMAP
 #include <sys/mman.h>
+#endif
 #include <sys/stat.h>
 #include <unistd.h>
 #endif
@@ -5064,7 +5076,7 @@ inline bool padded_string_builder::reserve(size_t additional) noexcept {
 }
 
 
-#ifndef _WIN32
+#if SIMDJSON_HAS_MMAP
 simdjson_inline padded_memory_map::padded_memory_map(const char *filename) noexcept {
 
     int fd = open(filename, O_RDONLY);
@@ -5112,7 +5124,7 @@ simdjson_inline simdjson::padded_string_view padded_memory_map::view() const noe
 simdjson_inline bool padded_memory_map::is_valid() const noexcept {
   return data != nullptr;
 }
-#endif // _WIN32
+#endif // SIMDJSON_HAS_MMAP
 
 } // namespace simdjson
 
